src/proof/alethe/alethe_post_processor.cpp

```diff
@@ -404,6 +478,394 @@ bool AletheProofPostprocessCallback::update(Node res,
+                          conclusion,
+                          conclusion,
+                          {children[i]},
+                          {},
+                          *cdp);
+            new_children[i] = conclusion;
+          }
+        }
+      }
+
+      // If res is not an or node, then it's necessarily a singleton clause.
+      bool isSingletonClause = res.getKind() != kind::OR;
+      // Otherwise, we need to determine if res if it's of the form (or t1 ...
+      // tn), corresponds to the clause (cl t1 ... tn) or to (cl (OR t1 ...
+      // tn)). The only way in which the latter can happen is if res occurs as a
+      // child in one of the premises, and is not eliminated afterwards. So we
+      // search for res as a subterm of some children, which would mark its last
+      // insertion into the resolution result. If res does not occur as the
+      // pivot to be eliminated in a subsequent premise, then, and only then, it
+      // is a singleton clause.
+      if (!isSingletonClause)
+      {
+        size_t i;
+        // Find out which child introduced res. There can be at most one by
+        // design of the proof production. After the loop finishes i is the
+        // index of the child C_i that introduced res. If i=0 none of the
+        // children introduced res as a subterm and therefore it cannot be a
+        // singleton clause.
+        for (i = children.size(); i > 0; --i)
+        {
+          // only non-singleton clauses may be introducing
+          // res, so we only care about non-singleton or nodes. We check then
+          // against the kind and whether the whole or node occurs as a pivot of
+          // the respective resolution
+          if (children[i - 1].getKind() != kind::OR)
+          {
+            continue;
+          }
+          size_t pivotIndex = (i != 1) ? 2 * (i - 1) - 1 : 1;
+          if (args[pivotIndex] == children[i - 1]
+              || args[pivotIndex].notNode() == children[i - 1])
+          {
+            continue;
+          }
+          // if res occurs as a subterm of a non-singleton premise
+          if (std::find(children[i - 1].begin(), children[i - 1].end(), res)
+              != children[i - 1].end())
+          {
+            break;
+          }
+        }
+
+        // If res is a subterm of one of the children we still need to check if
+        // that subterm is eliminated
+        if (i > 0)
+        {
+          bool posFirst = (i == 1) ? (args[0] == trueNode)
+                                   : (args[(2 * (i - 1)) - 2] == trueNode);
+          Node pivot = (i == 1) ? args[1] : args[(2 * (i - 1)) - 1];
+
+          // Check if it is eliminated by the previous resolution step
+          if ((res == pivot && !posFirst)
+              || (res.notNode() == pivot && posFirst)
+              || (pivot.notNode() == res && posFirst))
+          {
+            // We decrease i by one such that isSingletonClause is set to false
```

Comment:
```suggestion
            // We decrease i by one, since it could have been the case that i
            // was equal to children.size(), so that isSingletonClause is set to
            // false
```